pkg/sql/logictest/tests/local-vec-off/local-vec-off_test: TestLogic_advisory_lock failed

pkg/sql/logictest/tests/local-vec-off/local-vec-off_test.TestLogic_advisory_lock [failed](https://mesolite.cluster.engflow.com/invocations/default/b65b3e8b-9720-425d-9439-a4a0d6118983?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9sb2NhbC12ZWMtb2ZmOmxvY2FsLXZlYy1vZmZfdGVzdA==) on master @ [de42597506b573a36254b6d8860bb643edfee47f](https://github.com/cockroachdb/cockroach/commits/de42597506b573a36254b6d8860bb643edfee47f):


```
=== RUN   TestLogic_advisory_lock
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestLogic_advisory_lock848321727
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156124 (expected label: C-bug)
[20:18:52] setting distsql_workmem='29187B';
[20:18:52] --- done: /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock with config local-vec-off: 19 tests, 1 failures
--- total: 19 tests, 1 failures
    logic.go:4874: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_advisory_lock848321727
--- FAIL: TestLogic_advisory_lock (1.91s)
=== RUN   TestLogic_advisory_lock/pg_try_advisory_xact_lock
    logic.go:3280: 
         
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock:71: SELECT pg_try_advisory_xact_lock(1::INT4, 2::INT4)
        expected:
            true
            
        but found (query options: "") :
            false
            
--- FAIL: TestLogic_advisory_lock/pg_try_advisory_xact_lock (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_advisory_lock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63364

pkg/sql/logictest/tests/local-vec-off/local-vec-off_test.TestLogic_advisory_lock [failed](https://mesolite.cluster.engflow.com/invocations/default/7ca93ff7-205e-4f42-b4e8-440acfb22799?testReportRun=9&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9sb2NhbC12ZWMtb2ZmOmxvY2FsLXZlYy1vZmZfdGVzdA==) on master @ [23ae4cb712d56f3311f276405da416d4792404c5](https://github.com/cockroachdb/cockroach/commits/23ae4cb712d56f3311f276405da416d4792404c5):


```
=== RUN   TestLogic_advisory_lock
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestLogic_advisory_lock3376549978
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156124 (expected label: C-bug)
[05:03:09] setting distsql_workmem='90573B';
[05:03:09] --- done: /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock with config local-vec-off: 19 tests, 1 failures
--- total: 19 tests, 1 failures
    logic.go:5004: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_advisory_lock3376549978
--- FAIL: TestLogic_advisory_lock (1.89s)
=== RUN   TestLogic_advisory_lock/pg_try_advisory_xact_lock
    logic.go:3409: 
         
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-vec-off/local-vec-off_test_/local-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/advisory_lock:71: SELECT pg_try_advisory_xact_lock(1::INT4, 2::INT4)
        expected:
            true
            
        but found (query options: "") :
            false
            
--- FAIL: TestLogic_advisory_lock/pg_try_advisory_xact_lock (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_advisory_lock.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/logictest/testdata/logic_test/advisory_lock
+++ b/pkg/sql/logictest/testdata/logic_test/advisory_lock
@@ -67,6 +67,7 @@ true
 statement ok
 COMMIT
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock(1::INT4, 2::INT4)
 ----
@@ -169,6 +170,7 @@ user testuser
 statement ok
 BEGIN
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock(1::INT4, 2::INT4)
 ----
@@ -588,6 +590,7 @@ user testuser
 statement ok
 BEGIN
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock(950001)
 ----
@@ -649,6 +652,7 @@ user root
 statement ok
 BEGIN
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock(950101)
 ----
@@ -731,16 +735,19 @@ SET database = defaultdb
 statement ok
 BEGIN
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock(960001)
 ----
 true
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock_shared(960002)
 ----
 true
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock(960003::INT4, 960004::INT4)
 ----
@@ -920,6 +927,7 @@ user testuser
 statement ok
 BEGIN
 
+retry
 query B
 SELECT pg_try_advisory_xact_lock(990001)
 ----

